src/main/java/seedu/duke/parsers/ModHappyParser.java

```diff
@@ -73,13 +83,27 @@ private Parser getCommandParser(String commandWord) throws UnknownCommandExcepti
         case (LIST_COMMAND_WORD):
             return new ListParser();
         case (ADD_COMMAND_WORD):
-            return new AddParser();
+            switch (parsedCommand.get(ARGUMENT).split(SPACE)[0]) {
+            case TASK:
+                return new AddTaskParser();
+            case MODULE:
+                return new AddModuleParser();
+            default:
+                throw new UnknownCommandException();
+            }
         case (DELETE_COMMAND_WORD):
             return new DeleteParser();
         case (MARK_COMMAND_WORD):
             return new MarkParser();
         case (EDIT_COMMAND_WORD):
-            return new EditParser();
+            switch (parsedCommand.get(ARGUMENT).split(SPACE)[0]) {
+            case TASK:
+                return new EditTaskParser();
+            case MODULE:
+                return new EditModuleParser();
+            default:
+                throw new UnknownCommandException();
+            }
```

Comment:
Same suggestion as in ModHappyParser

Reply:
shifted the logic to abstract class EditParser